### PR TITLE
Add optional logits slicing support to GPT2LMHeadModel

### PR DESCRIPTION
**Pull request by abdulrehhhhman**
This PR adds support for an optional `logits_to_keep` argument in GPT2LMHeadModel.
It allows users to slice logits output for memory-efficient inference or analysis.

Example:
```python
outputs = model(**inputs, logits_to_keep=2)

### CODE REVIEW

**Comment by abdulrehhhhman** [comment]
Hi @huggingface-team 👋
This PR introduces an optional logits_to_keep argument for GPT2LMHeadModel.forward, enabling selective slicing of output logits for memory-efficient inference and analysis.
It is backward-compatible and passes the existing GPT-2 test suite locally.
Thanks for reviewing!

**Comment by Rocketknight1** [comment]
The two lines above already do this!

```python
slice_indices = slice(-logits_to_keep, None) if isinstance(logits_to_keep, int) else logits_to_keep
logits = self.lm_head(hidden_states[:, slice_indices, :])
```

**Comment by abdulrehhhhman** [comment]
Thanks for the clarification! That makes sense — I see it’s already handled by those lines.
I learned a lot from digging into GPT2LMHeadModel’s internals
